src/client/components/pages/RoomAdmin/__tests__/RoomAdminTable.test.tsx

```diff
@@ -21,6 +24,186 @@ import {
+    const buildingDropDown = renderResult.getByLabelText('The table will be filtered by selected building name');
+    const dropdownOptions = within(buildingDropDown)
+      .getAllByRole('option') as HTMLOptionElement[];
+    const buildingDropdownLabels = dropdownOptions
+      .map(({ textContent }) => textContent);
+    const buildingList = adminRoomsResponse.map((el) => el.building.name);
+    // slicing 'All' value as it's manually added default value not generated from adminRoomsResponse
+    strictEqual(
+      buildingDropdownLabels.slice(1).length === buildingList.length, true,
+      'building dropdown menu is not listing all available buildings'
+    );
+  });
+  context('when building dropdown value change', function () {
+    it('filters table data for selected value', async function () {
+      await wait(() => renderResult.getAllByRole('row').length > 1);
+      const [buildingDropDown] = renderResult.getAllByLabelText('The table will be filtered by selected building name');
+      fireEvent.change(buildingDropDown, { target: { value: 'Bauer Laboratory' } });
+      await wait(() => renderResult.getAllByRole('row').length > 1);
+      const rows = renderResult.getAllByRole('row');
+      const bodyRow = rows.filter((row) => (
+        within(row).queryAllByRole('columnheader').length === 0
+      ));
+      bodyRow.forEach((row) => {
+        strictEqual(
+          within(row).queryByText('Bauer Laboratory') !== null,
+          true,
+          'Bauer Laboratory is not in the table as expected'
+        );
+      });
+    });
+  });
+});
+describe('Room input filter', function () {
+  let getStub: SinonStub;
+  let renderResult: RenderResult;
+  beforeEach(function () {
+    getStub = stub(LocationAPI, 'getAdminRooms');
+    getStub.resolves(adminRoomsResponse);
+    renderResult = render(
+      <RoomAdmin />
+    );
+  });
+  it('defaults to null', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const roomFilterInput = renderResult.getByLabelText('The table will be filtered by selected room name', { exact: true }) as HTMLSelectElement;
+    strictEqual(roomFilterInput.value, '');
+  });
+  it('filters table data when room name is entered', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const [roomInput] = renderResult.getAllByLabelText('The table will be filtered by selected room name');
+    fireEvent.change(roomInput, { target: { value: '101A' } });
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const rows = renderResult.getAllByRole('row');
+    const bodyRow = rows.filter((row) => (
+      within(row).queryAllByRole('columnheader').length === 0
+    ));
+    bodyRow.forEach((row) => {
+      strictEqual(
+        within(row).queryByText('101A') !== null,
+        true,
+        '101A is not in the table as expected'
+      );
+    });
+  });
```

Comment:
I would also recommend defining a `testRoomName` here that references `adminRoomsResponse` instead of hardcoding `101A`.